semgrep-core/core/Caching.ml

```diff
@@ -0,0 +1,376 @@
+(*
+   Decorate a pattern and target ASTs to make the suitable for memoization
+   during matching.
+*)
+
+module MV = Metavariable
+
+open Printf
+open AST_generic
+
+let debug = false
```

Comment:
You can use the pfff/commons/logging.ml module.
git grep logger for example of use inside semgrep.